internal/pool/pool_single_test.go

```diff
@@ -0,0 +1,31 @@
+package pool_test
+
+import (
+	"context"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/go-pg/pg/v10/internal/pool"
+)
+
+var _ = Describe("SingleConnPool", func() {
+	var p *pool.SingleConnPool
+
+	BeforeEach(func() {
+		p = pool.NewSingleConnPool(nil, &pool.Conn{})
+	})
+
+	It("remove a conn due to context is cancelled", func() {
+		ctx, cl := context.WithCancel(context.TODO())
```

Comment:
I'd rename `cl` to `cancel` to make it clear here.

Reply:
Yes, renamed: ec32de71bf6b8cf79af72db3011ffa4572c753c4